torch/nn/parallel/data_parallel.py

```diff
@@ -78,17 +78,21 @@ def _to_cuda(obj):
 
         replicas = self.replicate(self.module, self.device_ids)
         scattered = self.scatter(inputs, self.device_ids)
-
+        used_gpus = len(scattered)  # The last GPU might not be used. For example, input of size 5, on 4 GPUs
```

Comment:
I don't think it's a good example because it could have been scattered as (2,1,1,1). 5 GPUs and 4 elements makes sense.